sw/device/tests/otbn_ecdsa_op_irq_test.c

```diff
@@ -74,6 +76,119 @@ static const otbn_addr_t kOtbnVarXR = OTBN_ADDR_T_INIT(p256_ecdsa, x_r);
 
 const test_config_t kTestConfig;
 
+/**
+ * The plic dif to access the hardware.
+ */
+static dif_rv_plic_t plic;
+
+/**
+ * The otbn context handler.
+ */
+static otbn_t otbn_ctx;
+
+/**
+ * The peripheral which fired the irq to be filled by the irq handler.
+ */
+static volatile top_earlgrey_plic_peripheral_t plic_peripheral;
+
+/**
+ * The irq id to be filled by the irq handler.
+ */
+static volatile dif_rv_plic_irq_id_t irq_id;
+
+/**
+ * The otbn irq to be filled by the irq handler.
+ */
+static volatile dif_otbn_irq_t irq;
+
+/**
+ * Provides external IRQ handling for otbn tests.
+ *
+ * This function overrides the default OTTF external ISR.
+ *
+ * It performs the following:
+ * 1. Claims the IRQ fired (finds PLIC IRQ index).
+ * 2. Compute the OTBN peripheral.
+ * 3. Compute the otbn irq.
+ * 4. Clears the IRQ at the peripheral.
+ * 5. Completes the IRQ service at PLIC.
+ */
+void ottf_external_isr(void) {
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kTopEarlgreyPlicTargetIbex0,
+                                     (dif_rv_plic_irq_id_t *)&irq_id));
+
+  plic_peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  irq = (dif_otbn_irq_t)(irq_id -
+                         (dif_rv_plic_irq_id_t)kTopEarlgreyPlicIrqIdOtbnDone);
+
+  CHECK_DIF_OK(dif_otbn_irq_acknowledge(&otbn_ctx.dif, irq));
+
+  // Complete the IRQ by writing the IRQ source to the Ibex specific CC.
+  // register.
+  CHECK_DIF_OK(
+      dif_rv_plic_irq_complete(&plic, kTopEarlgreyPlicTargetIbex0, irq_id));
+}
+
+static void otbn_wait_for_done_irq(otbn_t *otbn_ctx) {
+  // Clear the otbn irq variable: we'll set it in the interrupt handler when
+  // we see the Done interrupt fire.
+  irq = irq_id = plic_peripheral = UINT32_MAX;
```

Comment:
Uh. This is definitely too clever. Can you just assign each in a separate statement?